src/metabase/upload.clj

```diff
@@ -41,11 +41,25 @@
 
 (set! *warn-on-reflection* true)
 
+(def ^:private max-field-name-length
+  "This tracks the size of the metabase_field.name field"
+  254)
+
+(def ^:private unique-field-name-buffer
```

Comment:
I'm not sure this is worth it, as it only ends up supporting `create` and not `update` (for now).

I don't really like the idea of the buffer - it can truncate things it didn't need to, and could end up getting too long again. A better solution would be a smarter version of `uniquify-names` that preserves a maximum length. Not sure if it's better to extend that complex util or write something minimal for our needs.

Reply:
Decided to keep it for now. Can revisit if we work on the related bug.